action.yml

```diff
@@ -28,16 +32,44 @@ runs:
         key: dune-cache-${{ runner.os }}-${{ runner.arch }}
     - name: Resolve dependencies
       shell: bash
+      env:
+        SETUPDUNEDIR: ${{ inputs.directory }}
+      run: |
+        cd "$SETUPDUNEDIR" && dune pkg lock
+      if: inputs.automagic == 'true'
+    - name: Install depexts
+      shell: bash
+      env:
+        SETUPDUNEDIR: ${{ inputs.directory }}
+        OS: ${{ runner.os }}
       run: |
-        dune pkg lock
+        DEPEXTS="$(cd "$SETUPDUNEDIR" >/dev/null && dune show depexts 2>&1)"
```

Comment:
I suspect the command substitution construct silently captures any potential error exit code of `dune show depexts` and would like result in continuing the steps and then attempt to install the error message as packages.

Reply:
I don’t know if `dune show depexts` can fail but the exit code of the whole line is the exit code of `cd "$SETUPDUNEDIR" >/dev/null && dune show depexts 2>&1`, so this should work (famous last words, but that’s why I wrote it that way; I really hope that’s not broken on the ancient bash used on Macs).

Reply:
Ah, sorry I was confused - indeed GitHub actions set -e by default here, so I agree with you. However, I was reminded of some struggle I had while working on that part in https://github.com/ocaml-dune/setup-dune/pull/5 :

> I don’t know if `dune show depexts` can fail

With this command substitution pattern, in case of failure the output is lost and never shown to the user (not even in the logs I think). Which would make any troubleshoot difficult. I'm not sure I would go with that one tbh.